AzureCalling/app/src/main/java/com/azure/samples/communication/calling/views/activities/SignInActivity.java

```diff
@@ -0,0 +1,123 @@
+import com.azure.samples.communication.calling.R;
+import com.azure.samples.communication.calling.contracts.Constants;
+import com.azure.samples.communication.calling.externals.authentication.AADAuthHandler;
+import com.azure.samples.communication.calling.externals.authentication.UserProfile;
+import com.azure.samples.communication.calling.utilities.AppSettings;
+import com.microsoft.fluentui.widget.Button;
+
+public class SignInActivity extends AppCompatActivity {
+
+    private static final String LOG_TAG = SignInActivity.class.getSimpleName();
+
+    private Button signInButton;
+    private AADAuthHandler authHandler;
+    private AppSettings appSettings;
+
+    private SharedPreferences sharedPreferences;
+    private SharedPreferences.Editor editor;
+
+    @Override
+    protected void onCreate(final Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_signin);
+
+        initializeAuth();
+        initializeUI();
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+
+        if (!appSettings.isAADAuthEnabled()) {
+            navigateToIntroView();
+        }
+    }
+
+    @Override
+    public void onBackPressed() {
+        finishAffinity();
+    }
+
+    private void initializeUI() {
+        signInButton = findViewById(R.id.sign_in_button);
+        signInButton.setOnClickListener(l -> onClickSignInButton());
+
+        if (getSupportActionBar() != null) {
+            getSupportActionBar().hide();
+        }
+    }
+
+    private void initializeAuth() {
+        sharedPreferences = this.getSharedPreferences(Constants.ACS_SHARED_PREF, Context.MODE_PRIVATE);
+        editor = sharedPreferences.edit();
+
+        authHandler = ((AzureCalling) getApplication()).getAadAuthHandler();
+        appSettings = ((AzureCalling) getApplication()).getAppSettings();
+    }
+
+    private void toggleProgress(final boolean show) {
+        final View progressOverlay = findViewById(R.id.overlay_loading);
+        if (show) {
+            progressOverlay.setVisibility(View.VISIBLE);
+            signInButton.setText(R.string.signing_in);
+            signInButton.setEnabled(false);
+        } else {
+            progressOverlay.setVisibility(View.GONE);
+            signInButton.setText(R.string.sign_in);
+            signInButton.setEnabled(true);
+        }
+    }
+
+    private void navigateToIntroView() {
+        final Intent intent = new Intent(this, IntroViewActivity.class);
+        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+        toggleProgress(false);
+        startActivity(intent);
+    }
+
+    private void cacheProfile(final UserProfile profile) {
+        editor.putBoolean(IS_LOGGED_IN, true);
+        editor.putString(USERNAME, profile.getDisplayName());
+        editor.putString(GIVEN_NAME, profile.getDisplayName());
+        editor.putString(ACS_DISPLAY_NAME, profile.getDisplayName());
```

Comment:
Every time user will open app, the request will be make to make sure, user is signed in. I do not think we need to cache this info.

Reply:
No. this is only when user logs in. User do not have to log in every time the app is opened.